internal/rbd/rbd_journal.go

```diff
@@ -521,28 +521,45 @@ func undoVolReservation(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
 // complete omap mapping between imageName and volumeID.
 
 // RegenerateJournal performs below operations
+// Extract required parameters journalPool, pool from volumeAttributes
+// Extract optional parameter volumeNamePrefix from volumeAttributes
 // Extract information from volumeID
 // Get pool ID from pool name
 // Extract uuid from volumeID
 // Reserve omap data
 // Generate new volume Handler
 // The volume handler won't remain same as its contains poolID,clusterID etc
 // which are not same across clusters.
+// nolint:gocyclo // TODO: reduce complexity
```

Comment:
Can we have this for now, until `// TODO check clusterID mapping exists ` at line 568 is also addressed so we can get better idea of how this function can be refractored (or keep it the same).

IMO, having all these steps in one function will help readability & understandability (or am missing how to split it into helper functions :sweat: )

Reply:
just incase you are keeping `// nolint:gocyclo` do add cyclop linter too.

Reply:
>just incase you are keeping // nolint:gocyclo do add cyclop linter too.

I guess this `nolint` will also be removed in following prs but noted thanks!